runtime/src/main/kotlin/kotlin/native/internal/KTypeImpl.kt

```diff
@@ -21,6 +21,40 @@ internal class KTypeImpl(
     override fun hashCode(): Int {
         return (classifier?.hashCode() ?: 0) * 31 * 31 + this.arguments.hashCode() * 31 + if (isMarkedNullable) 1 else 0
     }
+
+    override fun toString(): String {
+        val classifierString = when (classifier) {
+            is KClass<*> -> classifier.qualifiedName ?: classifier.simpleName
```

Comment:
Does it handle `object` properly?

Reply:
What do you mean by `object` and how do you expect it to be handled?

Reply:
```diff --git a/backend.native/tests/codegen/ktype/ktype1.kt b/backend.native/tests/codegen/ktype/ktype1.kt
index a3d9fe3d4..aacb45ca6 100644
--- a/backend.native/tests/codegen/ktype/ktype1.kt
+++ b/backend.native/tests/codegen/ktype/ktype1.kt
@@ -17,17 +17,23 @@ class D
 fun <T> kTypeForCWithTypeParameter() = kType<C<T>>()
 
 class Outer<T> {
+    companion object Friend
     inner class Inner<S>
 }
 
+object Object
+
 private val pkg = "codegen.ktype.ktype1"
 
 @Test
+@UseExperimental(ExperimentalStdlibApi::class)
 fun test1() {
     assertEquals("$pkg.C<kotlin.Int?>", kType<C<Int?>>().toString())
     assertEquals("$pkg.C<$pkg.C<kotlin.Any>>", kType<C<C<Any>>>().toString())
 
     assertEquals("$pkg.C<kotlin.Any?>", kTypeForCWithTypeParameter<D>().toString())
+    assertEquals("$pkg.Object", kType<Object>().toString())
+    assertEquals("$pkg.Outer.Friend", kType<Outer.Friend>().toString())
 
     val innerKType = kType<Outer<D>.Inner<String>>()
     assertEquals(Outer.Inner::class, innerKType.classifier)
```

Reply:
Works for me.

Reply:
That's what I meant by `object` and it better be tested. Can we have an anonymous object construct somehow be passed as type parameter?

Reply:
Yes we can, it doesn't have string representation (`toString` returns `"(not denotable)"`).